source/reflect.h2

```diff
@@ -111,10 +122,13 @@ compiler_services: @polymorphic_base @copyable type =
 
         //  Now parse this single declaration from
         //  the lexed tokens
-        return parser.parse_one_declaration(
-            tokens*.get_map().begin()*.second,
-            generated_tokens*
-        );
+        ret = parser.parse_one_declaration(
+                tokens*.get_map().begin()*.second,
+                generated_tokens*
+              );
+        if !ret {
+            error( "parse failed - the source string is not a valid statement:\n(source)$");
```

Comment:
The `source` string view is modified.
You need to add an `original_source := source;` at the beginning.
This is the diagnostic for my previous bug:
```output
algorithms.test.cpp2: error: unexpected text '
lgorithms.test.cpp2: error: unexpected text '
lgorithms.test.cpp2: error: invalid else branch body (at '{')
algorithms.test.cpp2: error: ill-formed initializer (at '{')
algorithms.test.cpp2(37,1): error: while applying @array - parse failed - the source string is not a valid statement:
}
algorithms.test.cpp2(37,1): error: while applying @array - the provided source string is not a valid statement
```
Note how the source string is just `}`.